### ISSUE TITLE
BERT VS RoBERTa

### ISSUE THREAD

**Issue by zonghai-yao** [dmis-lab/BioLAMA#7]
Hi, very interesting work. I have a question about BERT VS RoBERTa. 

In original LAMA experiments, many people find BERT has a better performance than RoBERTa. However, I try the same thing in BioLAMA, and in my experiment results, RoBERTa is much better than BERT. I am not sure whether or not you tried the same thing (I do not see a related thing in your paper), if so, do you have the same result (Roberta >> BERT)

I try to explain it, and find one possible point: when I initialize the model with huggingface AutoModelWithLMHead, RoBERTa actually has everything from its checkpoint (it uses lm_head in the top), but BERT does not have 'cls.seq_relationship.weight', 'cls.seq_relationship.bias' in the head and need randomly init it. I think this definitely influences the BERT performance, right? Do you think this is a big issue? When you try BERT, did you see this warning, and how did you handle this or did you just ignore it?
warning: Some weights of the model checkpoint at bert-base-cased were not used when initializing BertForMaskedLM: ['cls.seq_relationship.weight', 'cls.seq_relationship.bias']

Thanks

**Comment by zonghai-yao** [comment]
Hi, please ignore my "possible point", I wrongly think it's "Some weights were not initialized from the model checkpoint" :). But, my RoBERTa bioLAMA result is still much better than BERT result, this still makes me confused, since it's very different from the LAMA story. So my question now is only "do you have the same result (Roberta >> BERT)"? Thanks

**Comment by mjeensung** [comment]
Hi, @seasonyao 

Thanks for reaching out to us.
Basically, Bio-LM described in our paper uses RoBERTa as a backbone language model, and we observed that Bio-LM outperforms BioBERT on BioLAMA. We believe that the main reason for this is attributed to the customized vocabulary that Bio-LM uses.
We haven't tried directly comparing BERT and RoBERTa on BioLAMA, but one possible reason I can think of is that the vocabulary set of RoBERTa contains more biomedical terms than the one of BERT. What do you think about this?

**Comment by seanswyi** [comment]
@seasonyao I don't recall seeing work that directly investigates the BERT > RoBERTa phenomenon, but I do recall that the authors of the [K-Adapter](https://aclanthology.org/2021.findings-acl.121/) model provide a possible explanation in the appendix: it all basically comes down to the tokenization algorithms and vocabularies of the two LMs.

Recall that BERT uses a character-level BPE in contrast to RoBERTa's byte-level BPE. Taking into account the advantages of a byte-level BPE, I assume that this difference allows RoBERTa to better encode longer entity names, which are prevalent in the biomedical domain, and thus results in higher performance on the BioLAMA benchmark dataset.

**Comment by zonghai-yao** [comment]
@seanswyi Thanks for the reply. I noticed BERT>RoBERTa phenomenon in LAMA from two papers, AUTOPROMPT and OptiPrompt. They both treat it as a surprise but did not explain too much about it. Of course, they mentioned "prompting is a lower bound on a model’s knowledge: the lower relative performance does not mean that the model actually knows less."

I think your explanation about longer entity names is very interesting and seems a good direction, worth to do some experiments to validate it. Thanks

**Comment by zonghai-yao** [comment]
@mjeensung yes, as you said I found Bio-LMs (RoBERTa bio family) outperforming BERT bio family (like BioBert, clinicalBERT). And in my experiments, Roberta itself has the same level of performance as BioBert, clinicalBERT, which caught my attention. Like you said, maybe the vocabulary set of RoBERTa contains more biomedical terms than the one of BERT, which may come from @seanswyi's assumption (RoBERTa to better encode longer entity names).

Actually, I have another question about the metric ACC@N. Some relation (like disease and symptoms --> covid19, cough, fever, tremble......) is N-M, for certain data like (covid19, tremble), during BioLAMA, if the top5 results of prompt "covid19 have symptoms such as [MASK]"  is "fever, cough, x1, x2, x3, x4, x5, x6, tremble", you will still treat (covid19, tremble) as correct in ACC@1 and ACC@5, right? As you mentioned in the paper, " if any of the top k object entities are included in the annotated object list", the object list here is "cough, fever, tremble......", so even though "tremble" is not in the top5, for relation (covid19, tremble), you still think it's correct. 

My question is, is such a setting fair? Why does not use the LAMA setting for the N-M metric --> "To account for multiple valid objects for a subject-relation pair (i.e., for N-M relations), we remove from the candidates when ranking at test time all other valid objects in the training data other than the one we test." --> from my understanding, this means they change "fever, cough, x1, x2, x3, x4, x5, x6, tremble" into "x1, x2, x3, x4, x5, x6, tremble", and will find "tremble" is not in the top5.

Let me know if I misunderstood anything, thanks!

**Comment by zonghai-yao** [comment]
@mjeensung You assume the vocabulary set is an important point here, so did you also compare different BioLM results in your experiments? I see there are models with and without training BPE vocab (like RoBERTa-base-PM and RoBERTa-base-PM-Voc). So is RoBERTa-base-PM-Voc result in BioLAMA better than RoBERTa-base-PM? Or is it the opposite?
